rework_pysatl_mpest/initializers/clusterize/score_functions.py

```diff
@@ -0,0 +1,32 @@
+import numpy as np
+
+from ...core import MixtureModel
+from ...distributions import ContinuousDistribution
+
+
+def _calculate_component_log_likelihood(model: ContinuousDistribution, X: np.ndarray, H_k: np.ndarray) -> float:
+    """Calculates the weighted log-likelihood for a single component"""
+    log_probs = np.clip(model.lpdf(X), -1e9, -1e-9)
+    return np.sum(H_k * log_probs)
+
+
+def _calculate_mixture_log_likelihood(model: MixtureModel, X: np.ndarray) -> float:
+    """Calculates the total log-likelihood for a mixture model"""
+    return np.sum(model.loglikelihood(X))
```

Comment:
`MixtureModel.loglikelihood` returns scalar, so:
```suggestion
    return model.loglikelihood(X)